### PR TITLE
Update observability.json.adoc

### PR DESCRIPTION
**Pull request by dockerymick**
https://github.com/open-cluster-management/backlog/issues/12943

### CODE REVIEW

File: apis/observability.json.adoc [hunk 1]
```diff
@@ -121,18 +124,38 @@ __required__|Parameters describing the MultiClusterObservability resource to be
...
       }
-      "receiveStorageSize":
-      "ruleStorageSize":
-      "storageClass":
-      "storeStorageSize":
+    "storageConfig": {
+      "alertmanagerStorageSize":
+      "compactStorageSize":
+      "metricObjectStorage": {
+      "name": "thanos-object-storage",
```
Comment (dockerymick) [inline apis/observability.json.adoc]: This link does not work: https://thanos.io/storage.md/#configuration

File: apis/observability.json.adoc [hunk 2]
```diff
@@ -359,9 +363,116 @@ __optional__|The amount of storage applied to thanos store stateful sets. Defaul
 |===
 |Name|Description|Schema
 |**key** +
-__required__|The key of the secret to select from. Must be a valid secret key.|string
+__required__|The key of the secret to select from. Must be a valid secret key. See NEED LINK currently broken|string
```
Comment (dockerymick) [inline apis/observability.json.adoc]: Link from https://github.com/open-cluster-management/multicluster-observability-operator/blob/main/docs/MultiClusterObservability-CRD.md is broken

File: apis/observability.json.adoc [hunk 3]
```diff
@@ -107,9 +107,12 @@ __required__|Parameters describing the MultiClusterObservability resource to be
     "enableDownsampling":
     "imagePullPolicy":
     "imagePullSecret":
+    "nodeSelector":
     "observabilityAddonSpec":{
       "enableMetrics":
       "interval":
+    "resources":
+    "replicas":
```
Comment (clyang82) [inline apis/observability.json.adoc]: there is no `replicas` here. need 2 space indent for `"resources"`

File: apis/observability.json.adoc [hunk 4]
```diff
@@ -359,9 +363,116 @@ __optional__|The amount of storage applied to thanos store stateful sets. Defaul
...
+**advanced**
+
+[options="header", cols=".^2a,.^3a,.^4a"]
+|===
+|Name|Description|Schema
+|**retentionConfig** +
+__required__|Specifies the data retention configuration to be used by observability.|`RetentionConfig`
+|**rbacQueryProxy** +
+__optional__|Specifies the replicas and resources for the rbac-query-proxy deployment.|CommonSpec
+|**grafan** +
```
Comment (clyang82) [inline apis/observability.json.adoc]: s/grafan/grafana

File: apis/observability.json.adoc [hunk 5]
```diff
@@ -359,9 +363,116 @@ __optional__|The amount of storage applied to thanos store stateful sets. Defaul
...
+__optional__|Specifies the replicas and resources for the grafana deployment|CommonSpec
+|**alertmanager** +
+__optional__|Specifies the replicas and resources for alertmanager statefulset.|CommonSpec
+|**observatoriumAPI** +
+__optional__|Specifies the replicas and resources for the `observatorium-api` deployment.|CommonSpec
+|**queryFrontend +
```
Comment (clyang82) [inline apis/observability.json.adoc]: missed `**`

File: apis/observability.json.adoc (hunk 1 above)
Comment (clyang82) [reply apis/observability.json.adoc]: use this one https://thanos.io/tip/thanos/storage.md/

File: apis/observability.json.adoc (hunk 2 above)
Comment (clyang82) [reply apis/observability.json.adoc]: it is available.

**Comment by clyang82** [comment]
some format error if you view the file

File: apis/observability.json.adoc [hunk 6]
```diff
@@ -121,18 +124,38 @@ __required__|Parameters describing the MultiClusterObservability resource to be
...
-      "storageClass":
-      "storeStorageSize":
+    }
+    "receiveStorageSize":
+    "ruleStorageSize":
+    "storageClass":
+    "storeStorageSize":
     }
   }
 }
```
Comment (clyang82) [inline apis/observability.json.adoc:114]: Suggest to replace ^^ request body with
```
{
  "apiVersion": "observability.open-cluster-management.io/v1beta2",
  "kind": "MultiClusterObservability",
  "metadata": {
    "name": "example"
  },
  "spec": {
    "observabilityAddonSpec": {}
    "storageConfig": {
      "metricObjectStorage": {
        "name": "thanos-object-storage",
        "key": "thanos.yaml"
       }
   }
}

**Review by clyang82 — CHANGES_REQUESTED** [review]

File: apis/observability.json.adoc (hunk 4 above)
Comment (dockerymick) [reply apis/observability.json.adoc]: great catch

File: apis/observability.json.adoc (hunk 5 above)
Comment (dockerymick) [reply apis/observability.json.adoc]: Thanks!

**Comment by dockerymick** [comment]
@clyang82 thanks for your review and feedback. When you have an opportunity, may you rereview the PR?

File: apis/observability.json.adoc [hunk 7]
```diff
@@ -291,44 +267,25 @@ __required__|<<_rhacm-docs_apis_multiclusterobservability_jsonmulticlusterobserv
...
 __optional__| Enable or disable the downsample. Default value is `true`. If there is no downsample data, the query is unavailable.| boolean
 |**imagePullPolicy** +
-__optional__| Pull policy for the Observability images.|string
+__optional__| Pull policy for the MultiClusterObservability images. The default value is `Always`.|corev1.PullPolicy
 |**imagePullSecret** +
-__optional__| Pull secret for the Observability images.|string
-|**nodeSelector** +
-__optional__|Supports node selection constraints|object
+__optional__| Pull secret for the MultiClusterObservability images. The default value is `multiclusterhub-operator-pull-secret`|string
+**nodeSelector** +
```
Comment (clyang82) [inline apis/observability.json.adoc]: need start with `|`

File: apis/observability.json.adoc [hunk 8]
```diff
@@ -359,9 +316,116 @@ __optional__|The amount of storage applied to thanos store stateful sets. Defaul
...
+|===
+|Name|Description|Schema
+|**resources** + 
+_optional_|Compute resources required by MultiCLusterObservability.|corev1.ResourceRequirements
+|**replicas** +
+__optional__|Replicas for MultiClusterObservability.|integer
+|**memoryLimitMb** +
+__optional__|Memory limit of Memcached in megabytes.|integer
+|**maxItemSize** +
+__optional__|Max item size of Memcached. The default value is `1m, min:1k, max:1024m`.
```
Comment (clyang82) [inline apis/observability.json.adoc]: append `|string`

File: apis/observability.json.adoc [hunk 9]
```diff
@@ -359,9 +316,116 @@ __optional__|The amount of storage applied to thanos store stateful sets. Defaul
...
+|**connectionLimit** +
+__optional__|Max simultaneous connections of Memcached.|integer
```
Comment (clyang82) [inline apis/observability.json.adoc]: missed default value here.

File: apis/observability.json.adoc (hunk 7 above)
Comment (dockerymick) [reply apis/observability.json.adoc]: Great catch, thanks!

File: apis/observability.json.adoc (hunk 9 above)
Comment (dockerymick) [reply apis/observability.json.adoc]: @clyang82 the default value wasn't listed in the table for `connectionLimit`. 

https://github.com/open-cluster-management/multicluster-observability-operator/blob/main/docs/MultiClusterObservability-CRD.md#cacheconfig

Comment (clyang82) [reply apis/observability.json.adoc]: Gotcha. we do not set the default value. it is set by observatorium-operator. I am OK w/o default value here. Thanks.

**Review by clyang82 — APPROVED** [review]
/lgtm

**Pull request merged**